### PR TITLE
Exclude GET verb from LroExtension rule validation

### PR DESCRIPTION
**Pull request by AkhilaIlla**
https://github.com/Azure/azure-openapi-validator/issues/549

This PR contains updates to
RPC-GET-V1-01(GetOperation200): to allow GET with 202 response code and 
RPC-Post-V1-09(LroExtension): to exclude GET from x-ms-long-running-operation presence validation
in order to support pooling kinda API's

```
"/providers/Microsoft.Subscription/subscriptionOperations/{operationId}": {
      "get": {
        "description": "Get the status of the pending Microsoft.Subscription API operations.",
        "operationId": "SubscriptionOperation_Get",
        "x-ms-examples": {
          "getPendingSubscriptionOperations": {
            "$ref": "./examples/getSubscriptionOperation.json"
          }
        },
        "responses": {
          "200": {
            "description": "Successful completion of the asynchronous operation",
            "schema": {
              "$ref": "#/definitions/SubscriptionCreationResult"
            }
          },
          "202": {
            "description": "Accepted. Subscription update is in progress.",
            "headers": {
              "Location": {
                "description": "The URL where the status of the asynchronous operation can be checked.",
                "type": "string"
              },
              "Retry-After": {
                "description": "The amount of delay to use while the status of the operation is checked. The value is expressed in seconds.",
                "type": "integer",
                "format": "int64"
              }
            }
          },
          "default": {
            "description": "Error response describing why the operation failed.",
            "schema": {
              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
            }
          }
        },
        "parameters": [
          {
            "name": "operationId",
            "in": "path",
            "description": "The operation ID, which can be found from the Location field in the generate recommendation response header.",
            "required": true,
            "type": "string"
          },
          {
            "$ref": "#/parameters/apiVersionParameter"
          }
        ]
      }
    },
```

### CODE REVIEW

File: docs/get-response-codes.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+## Applies to
+
+ARM OpenAPI(swagger) specs
+
+## Related ARM Guideline Code
+
+- RPC-Get-V1-01
+
+## Output Message
+
```
Comment (rkmanda) [inline docs/get-response-codes.md:16]: Lets use the new template that was decided for the help files.

Comment (rkmanda) [reply docs/get-response-codes.md:16]: cant find the link where this was documented. Please check with Tejaswi

File: docs/get-response-codes.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+## CreatedAt
+
+July 07, 2022
+
+## LastModifiedAt
+
+July 07, 2022
+
+## How to fix the violation
+
```
Comment (rkmanda) [inline docs/get-response-codes.md:34]: include good and bad examples

File: common/changes/@microsoft.azure/openapi-validator-rulesets/akhilailla-fix_RPC-Post-V1-09_and_RPC-Get-V1-01_2023-11-02-15-20.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft.azure/openapi-validator-rulesets",
+      "comment": "Update RPC-GET-V1-01(GetOperation200) to allow GET with 202 response code and RPC-Post-V1-09(LroExtension) to exclude GET from x-ms-long-running-operation presence validation",
+      "type": "patch"
```
Comment (rkmanda) [inline common/changes/@microsoft.azure/openapi-validator-rulesets/akhilailla-fix_RPC-Post-V1-09_and_RPC-Get-V1-01_2023-11-02-15-20.json:6]: "comment": "Update RPC-GET-V1-01(GetOperation200) to allow GET with 202 response code if the GET represents the location header polling url. Update RPC-Post-V1-09(LroExtension) to exclude GET from x-ms-long-running-operation presence validation",

**Comment by rkmanda** [comment]
>             description: "Operations with a 202 response must specify `x-ms-long-running-operation: true`.",

Make it explicit everywhere that this does not apply to Get including the help file

---
Refers to: packages/rulesets/generated/spectral/az-arm.js:891 in 290a9a6. [](commit_id = 290a9a6f31d9474a343fc745328dbb4318136f47, deletion_comment = False)

File: packages/rulesets/src/spectral/az-arm.ts [hunk 4]
```diff
@@ -302,16 +302,18 @@ const ruleset: any = {
     },
     // github issue https://github.com/Azure/azure-openapi-validator/issues/331
     // Get operation should return 200
-    // already have rule to check if operation returns non 2XX, it should mark it as 'x-ms-error-response' explicitly,
-    // so here on check if the 200 return '201','202','203'
+    // already have rule to check if operation returns non 2XX, it should mark it as 'x-ms-error-response' explicitly
+    // https://github.com/Azure/azure-openapi-validator/issues/549 
+    // GET can return 202 only if it is a polling action & has Location header defined. LroLocationHeader rule already checks if 202 response has Location header
+    // so here just check for non 200, 202 response codes i.e, '201','203','204'
     // RPC Code: RPC-Get-V1-01
```
Comment (rkmanda) [inline packages/rulesets/src/spectral/az-arm.ts:309]: why not make it an inclusion list for 200, 202 instead of an exclusion list for 201, 203, 204?

Comment (rkmanda) [reply packages/rulesets/src/spectral/az-arm.ts:309]: That way we get broader coverage

**Review by rkmanda — CHANGES_REQUESTED** [review]
:clock1:

File: common/changes/@microsoft.azure/openapi-validator-rulesets/akhilailla-fix_RPC-Post-V1-09_and_RPC-Get-V1-01_2023-11-02-15-20.json (hunk 3 above)
Comment (AkhilaIlla) [reply common/changes/@microsoft.azure/openapi-validator-rulesets/akhilailla-fix_RPC-Post-V1-09_and_RPC-Get-V1-01_2023-11-02-15-20.json:6]: I didnt add
"if the GET represents the location header polling url" cause that isnt being checked in this rule

File: packages/rulesets/src/spectral/az-arm.ts (hunk 4 above)
Comment (AkhilaIlla) [reply packages/rulesets/src/spectral/az-arm.ts:309]: thats a good point!
But we'd not be able to use the core function

File: docs/get-response-codes.md (hunk 1 above)
Comment (AkhilaIlla) [reply docs/get-response-codes.md:16]: [https://github.com/Azure/azure-openapi-validator/blob/main/CONTRIBUTING.md#how-to-write-a-new-validation-rule-using-typescript](https://github.com/Azure/azure-openapi-validator/blob/main/CONTRIBUTING.md#how-to-write-a-new-validation-rule-using-typescript)

**Review by rkmanda — APPROVED** [review]
:shipit:

**Pull request merged**